web/pdf_page_view.js

```diff
@@ -653,35 +664,33 @@ class PDFPageView {
             textLayer.setTextContentStream(readableStream);
             textLayer.render();
           }
+
+          if (this._annotationCanvasMap) {
+            if (!this.annotationLayer) {
+              this.annotationLayer =
+                this.annotationLayerFactory.createAnnotationLayerBuilder(
+                  div,
+                  pdfPage,
+                  /* annotationStorage = */ null,
+                  this.imageResourcesPath,
+                  this._annotationMode === AnnotationMode.ENABLE_FORMS,
+                  this.l10n,
+                  /* enableScripting = */ null,
+                  /* hasJSActionsPromise = */ null,
+                  /* mouseState = */ null,
+                  /* fieldObjectsPromise = */ null,
+                  /* annotationCanvasMap */ this._annotationCanvasMap
+                );
+            }
+            this._renderAnnotationLayer();
+          }
```

Comment:
Really sorry about the back-and-forth here, but seeing the latest round of changes it's probably overall simpler and more readable if we just do:
```suggestion
          if (this.annotationLayer) {
            this._renderAnnotationLayer();
          }
```